packages/ag-charts-enterprise/src/features/annotations/annotations.ts

```diff
@@ -134,26 +135,19 @@ class AxesButtons {
 }
 
 export class Annotations extends _ModuleSupport.BaseModuleInstance implements _ModuleSupport.ModuleInstance {
-    // TODO: When the 'restore-annotations' event is triggered from `ActionsOnSet.newValue()`, the module is still
-    // disabled when `onRestoreAnnotations()` is called, preventing the state from being restored. However,
-    // when `ObserveChanges()` is first called `target.enabled === false`, rather than `undefined`. So
-    // there is no way to detect if the module was actively disabled. This flag simulates a combined
-    // behaviour of both and is toggled when the module is actively disabled and enabled.
-    private __hackWasDisabled = false;
-
-    @_ModuleSupport.ObserveChanges<Annotations, boolean>((target, enabled) => {
+    @ObserveChanges<Annotations>((target, newValue?: boolean, oldValue?: boolean) => {
         const {
             ctx: { annotationManager, stateManager, toolbarManager },
         } = target;
 
-        toolbarManager.toggleGroup('annotations', 'annotations', { visible: Boolean(enabled) });
+        if (newValue === oldValue) return;
```

Comment:
Turns out the issue was simply that `ObserveChanges()` gets called twice on initial set with `{ newValue: true, oldValue: undefined }` then `{ newValue: true, oldValue: true }`.

Skipping this second call fixes the issue, much like `ActionOnSet()` but that is a transformer which gets called _before_ the value has changed, while the observers get called _after_ the value has changed.

I thought about adding an `ActionAfterSet()` as an observer, but didn't seem necessary for just this one case. Though it may be worth analysing our uses of `ObserveChanges()` to see if they can be improved similarly to prevent excessive calls and then use something like `ActionAfterSet()`.